api/src/controllers/listing.controller.ts

```diff
@@ -180,9 +181,27 @@ export class ListingController {
     return await this.listingService.process();
   }
 
+  @Put('lotteryStatus')
```

Comment:
I'm ok with this pattern and if you/the team like it better lets stick with this. But typically in a put endpoint like this I would have the listing id as a path parameter. So it would look like `@Put('lotteryStatus/:listingId')` and then update the ListingLotteryStatus dto to just have the new status

Reply:
@YazeedLoonat recommended we don't do that. Yazeed can you explain why that was again because I can't remember exactly what you said

Reply:
Adding a bit more context @ludtkemorgan :
- if you look at endpoints that currently take ids via the path params we almost always also pass the id via the dto and then in the service we only use the value in the dto

- in nestjs we can validate those path params to make sure they are uuid's but to do that everywhere would probably get messy. Whereas when doing that via the dto method we can leverage the already built in field validation

- this is minor but the path params make it so that the order the endpoints show up in controller matter